pkg/devfile/parser/data/2.1.0/components.go

```diff
@@ -0,0 +1,53 @@
+package version210
+
+import (
+	"strings"
+
+	"github.com/openshift/odo/pkg/devfile/parser/data/common"
+)
+
+// GetComponents returns the slice of DevfileComponent objects parsed from the Devfile
+func (d *Devfile210) GetComponents() []common.DevfileComponent {
+	return d.Components
+}
+
+// GetCommands returns the slice of DevfileCommand objects parsed from the Devfile
+func (d *Devfile210) GetCommands() []common.DevfileCommand {
+	var commands []common.DevfileCommand
+
+	for _, command := range d.Commands {
+		// we convert devfile command id to lowercase so that we can handle
+		// cases efficiently without being error prone
+		// we also convert the odo push commands from build-command and run-command flags
+		command.Exec.Id = strings.ToLower(command.Exec.Id)
+		commands = append(commands, command)
+	}
+
+	return commands
+}
+
+// GetParent returns the  DevfileParent object parsed from devfile
+func (d *Devfile210) GetParent() common.DevfileParent {
+	return d.Parent
+}
+
+// GetProjects returns the DevfileProject Object parsed from devfile
+func (d *Devfile210) GetProjects() []common.DevfileProject {
+	return d.Projects
+}
+
+// GetMetadata returns the DevfileMetadata Object parsed from devfile
+func (d *Devfile210) GetMetadata() common.DevfileMetadata {
+	return d.Metadata
+}
+
+// GetEvents returns the Events Object parsed from devfile
+func (d *Devfile210) GetEvents() common.DevfileEvents {
+	return d.Events
+}
+
+// GetAliasedComponents returns the slice of DevfileComponent objects that each have an alias
+func (d *Devfile210) GetAliasedComponents() []common.DevfileComponent {
```

Comment:
The comment here doesn't seem to reflect what it's doing?

It's returning the same thing as `GetComponents`

Reply:
Thanks for the comment @bigkevmcd 
This file was essentially copied from the odo deploy PR (https://github.com/openshift/odo/pull/3478) that this PR is based on, and modified for devfile v2.1.0
In devfile v1.0.0 the "name" attribute was not mandatory for the components defined, but in v2.0.0 (That this file was copied from), the "name" attribute has been made mandatory, which I think we are currently following in 2.1.0 as well
as a result, all components in v2.0.0 and v2.1.0 are aliased by default, rendering the ```GetAliasedComponents``` redundant for these versions, but the ```GetAliasedComponents``` function is defined in the ```DevfileData``` interface, which is probably why it needs to be implemented regardless